src/network/live_configuration/live_configuration.rs

```diff
@@ -0,0 +1,315 @@
+            // Blocked in socket.read_msg()
+            // Expect, we get a configuration and are Online / Running state
+            let config_result = live_config.get_configuration();
+            assert!(matches!(config_result, Ok(_)), "{:?}", config_result);
+            let thread_state = live_config.get_thread_status();
+            assert!(matches!(thread_state, ThreadStatus::Running));
+            let current_mode = live_config.get_current_mode();
+            assert!(matches!(current_mode, Ok(CurrentMode::Online)));
+            config_result.unwrap()
+        };
+
+        {
+            // Send a heartbeat via the websocket.
+            read_msg_tx
+                .send(tungstenite::Message::text(SERVER_HEARTBEAT))
+                .unwrap();
+            // Wait for thread to do some work and then to wait on websocket
+            read_msg_ping_rx.recv().unwrap();
+
+            // Expect no change due to heartbeat:
+            let config_result = live_config.get_configuration();
+            assert!(config_result.is_ok());
+            assert_eq!(config_result.unwrap(), config);
+            let thread_state = live_config.get_thread_status();
+            assert!(matches!(thread_state, ThreadStatus::Running));
+            let current_mode = live_config.get_current_mode();
+            assert!(matches!(current_mode, Ok(CurrentMode::Online)));
+        }
+
+        {
+            // Send any message via the websocket (it will be interpreted as new config is available).
+            read_msg_tx.send(tungstenite::Message::text("")).unwrap();
+            // Send the new configuration
+            let configuration = configuration_property1_enabled();
+            get_configuration_tx.send(configuration).unwrap();
+            // Wait for thread to do some work and then to wait on websocket
+            read_msg_ping_rx.recv().unwrap();
+
+            // Expect new configuration, and still running/online
+            let config_result = live_config.get_configuration();
+            assert!(config_result.is_ok());
+            assert_ne!(config_result.unwrap(), config);
+            let thread_state = live_config.get_thread_status();
+            assert!(matches!(thread_state, ThreadStatus::Running));
+            let current_mode = live_config.get_current_mode();
+            assert!(matches!(current_mode, Ok(CurrentMode::Online)));
+        }
+
+        {
+            // When the client is dropped, the thread will be finished
+            drop(live_config);
+
+            read_msg_tx
+                .send(tungstenite::Message::text(SERVER_HEARTBEAT))
+                .unwrap();
+
+            let r = read_msg_ping_rx.recv();
+            assert!(matches!(r, Err(RecvError { .. })), "{:?}", r)
+        }
+    }
+
+    #[test]
+    fn test_wait_for_initial_configuration() {
+        // TODO (or not): A way to create a LiveConfiguration object and wait until the first Configuration is available
+    }
```

Comment:
Remove. This is probably a different feature.

Reply:
I've created this issue: https://github.com/IBM/appconfiguration-rust-sdk/issues/67